lib/rules/new-line-between-multi-line-property.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * @fileoverview Enforce new lines between multi-line properties in Vue components.
+ * @author IWANABETHATGUY
+ */
+'use strict'
+
+const utils = require('../utils')
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+// @ts-ignore
+module.exports = {
+  meta: {
+    type: 'layout',
+    docs: {
+      description:
+        'enforce new lines between multi-line properties in Vue components',
+      categories: undefined,
+      url:
+        'https://eslint.vuejs.org/rules/new-line-between-multi-line-property.html'
+    },
+    fixable: 'whitespace', // or "code" or "whitespace"
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          // number of line you want to insert after multi-line property
+          minLineOfMultilineProperty: {
+            type: 'number',
+            minimum: 2
+          }
+        }
+      }
+    ]
+  },
+
+  /** @param {RuleContext} context */
+  create(context) {
+    // always insert one line
+    const insertLine = 1
+
+    let minLineOfMultilineProperty = 2
+    if (
+      context.options &&
+      context.options[0] &&
+      context.options[0].minLineOfMultilineProperty
+    ) {
+      minLineOfMultilineProperty = context.options[0].minLineOfMultilineProperty
+    }
+
+    /** @type {any[]} */
+    const callStack = []
+    /** @type {any[]} */
+    let comments = []
+    return Object.assign(
+      {
+        /**
+         * @param {Program} node
+         */
+        Program(node) {
+          comments = node.comments
+        }
+      },
+      utils.defineVueVisitor(context, {
+        CallExpression(node) {
+          callStack.push(node)
+        },
+        'CallExpression:exit'() {
+          callStack.pop()
+        },
+
+        /**
+         * @param {ObjectExpression} node
+         */
+        ObjectExpression(node) {
+          if (callStack.length) {
+            return
+          }
+          const properties = node.properties
+          for (let i = 1; i < properties.length; i++) {
+            const cur = properties[i]
+            const pre = properties[i - 1]
+
+            /** @type {any[]} */
+            const leadingComments = []
+            // getFirstLeadingComments is enough , after that break the loop.
+            for (
+              let commentsIndex = 0;
+              commentsIndex < comments.length;
+              commentsIndex++
+            ) {
+              const comment = comments[commentsIndex]
+              if (
+                comment.range[0] > pre.range[1] &&
+                comment.range[1] < cur.range[0]
+              ) {
+                leadingComments.push(comments[commentsIndex])
```

Comment:
(I haven't checked everything yet)
It's probably more efficient to use the SourceCode API.

https://eslint.org/docs/developer-guide/working-with-rules#contextgetsourcecode-1